elisp/ghc-check.el

```diff
@@ -126,13 +126,13 @@ nil            does not display errors/warnings.
       (setq mode-line-process " failed")
       (force-mode-line-update)))))
 
-(defun ghc-to-info (errs)
+(defun ghc-to-info (dirname errs)
   ;; [^\t] to include \n.
-  (let ((regex "^\\([^\n]*\\):\\([0-9]+\\):\\([0-9]+\\): *\\([^\t]+\\)")
+  (let ((regex "^.*/\\([^\n]*\\):\\([0-9]+\\):\\([0-9]+\\): *\\([^\t]+\\)")
```

Comment:
Could you explain this change in more detail please, using some example to show how the old regex is wrong.